compiler/src/iree/compiler/API/Internal/CompilerDriver.cpp

```diff
@@ -169,6 +170,57 @@ class rt_aligned_string_ostream : public llvm::raw_ostream {
   }
 };
 
+llvm::ThreadPoolStrategy getGlobalThreadPoolStrategy() {
+  // We allow two environment variables to control the compiler thread pool.
+  //   IREE_COMPILER_THREAD_COUNT: If present (and not empty or 0), will
+  //     create a thread pool with the requested number of threads, even
+  //     if this is unwise or beyond the limit limit of hardware concurrency.
+  //   IREE_COMPILER_TASK_COUNT: Specifies a target maximum number of
+  //     concurrent tasks to support at any given time. The actual number
+  //     of threads will be limited to the hardware concurrency if in
+  //     excess. If zero, then the hardware concurrency is used.
+  // In the absence of any environment variables, the behavior is as if
+  // IREE_COMPILER_TASK_COUNT was specified with a hard-coded value that
+  // has been chosen by project developers. If a number greater than
+  // zero IREE_COMPILER_THREAD_COUNT will take precedence.
+  const char *envThreadCount = getenv("IREE_COMPILER_THREAD_COUNT");
+  const char *envTaskCount = getenv("IREE_COMPILER_TASK_COUNT");
+
+  // As of 2023-11-11, the compiler was capable of exploiting ~12x parallelism
```

Comment:
Can we mention the specs of the machine as well? Just the number of available cores / threads should be enough.

Reply:
Done